File: Leetcode/arrays/ 219. Contains Duplicate II/solution.java

import java.util.HashSet;
import java.util.*;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Set<Integer> hashset = new HashSet<>();
        for (int i = 0; i < nums.length; i++) {
            if (hashset.contains(nums[i]))
                return true;
            hashset.add(nums[i]);

            if (hashset.size() > k)
                hashset.remove(nums[i - k]);
        }

        return false;

    }

    public static void main(String args[]) {

        Solution sol = new Solution();
        int[] nums = { 1, 2, 3, 1 };

        System.out.println("Output: " + sol.containsNearbyDuplicate(nums, 1));
    }
}

File: Leetcode/arrays/ 219. Contains Duplicate II/solution.cpp
#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

class Solution
{
public:
    bool containsNearbyDuplicate(vector<int> &nums, int k)
    {
        unordered_map<int, int> mp;
        int n = nums.size();

        for (int i = 0; i < n; i++)
        {
            if (mp.count(nums[i]))
            {
                if (abs(i - mp[nums[i]]) <= k)
                    return true;
            }
            mp[nums[i]] = i;
        }
        return false;
    }
};

int main()
{
    Solution sol;
    vector<int> nums = {1, 2, 3, 1};
    int k = 3;
    cout << sol.containsNearbyDuplicate(nums, k) << endl; 
    return 0;
}